src/Providers/OEmbed/N500px.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Embed\Providers\OEmbed;
+use Embed\Url;
+
+/**
+ * Class N500px
+ * 500px.com embeds.
+ */
+class N500px extends OEmbedImplementation
+{
+    /**
+     * {@inheritdoc}
+     */
+    public static function getEndPoint(Url $url)
+    {
+    	$matches=array();
+    	preg_match('#(https?://500px.com/photo/\d+/)#si', $url->getUrl(),$matches);
+    	
+    	if (isset($matches[1]) && $matches[1]) {
+    		return $matches[1].'/oembed.json';
+    	}	
```

Comment:
more correct will be 
````php
       if (is_numeric($this->request->getDirectoryPosition(1))) {         
           return $this->request->createUrl()->withDirectoryPosition(2,'oembed.json');
        }    	
````
because file can be like this https://500px.com/photo/109843523/somewhere-in-the-desert-by-catherine- and withAddedPath will make last element a dir. that won't work :)